app/views.py

```diff
@@ -0,0 +1,228 @@
+            return {"message": "no blank fields allowed"}, 400
+        if name.isspace():
+            return {"message": "name is invalid"}, 400
+        bucket = Bucketlist(name=name, user_id=g.user.id)
+        save(bucket)
+        msg = ("bucketlist -" + bucket.name + "- has been successfully "
+               "created with ID: " + str(bucket.id))
+        return {"message": msg}, 201
+
+    @marshal_with(bucketlistformat)
+    def get(self, id=None):
+        search = request.args.get("q") or None
+        page = request.args.get("page") or 1
+        limit = request.args.get("limit") or 20
+        if id:
+            bucketlist = Bucketlist.query.filter_by(id=id).first()
+            if not bucketlist or (bucketlist.user_id != g.user.id):
+                abort(404, "bucketlist not found")
+            return bucketlist, 200
+
+        if search:
+            bucket_search = Bucketlist.query.filter(Bucketlist.name.ilike(
+                "%" + search + "%")).filter_by(user_id=g.user.id).paginate(
+                int(page), int(limit), False)
+            if len(bucket_search.items) == 0:
+                abort(404, "bucketlist of that name not found")
+            else:
+                bckt = [bckt for bckt in bucket_search.items]
+                return bckt, 200
+
+        if page or limit:
+            bucket_collection = Bucketlist.query.filter_by(
+                user_id=g.user.id).paginate(int(page), int(limit), False)
+            bucket_disp = [bckt for bckt in bucket_collection.items]
+            return bucket_disp, 200
+
+        bucketlists = Bucketlist.query.filter_by(user_id=g.user.id).all()
+        return bucketlists, 200
+
+    def put(self, id):
+        self.reqparse.add_argument("name", type=str, required=True,
+                                   location="json",
+                                   help="bucketlist name required")
+        args = self.reqparse.parse_args()
+        name = args["name"]
+        if not is_not_empty(name):
+            return {"message": "no blank fields allowed"}, 400
+        if name.isspace():
+            return {"message": "name is invalid"}, 400
+        bucketlist = Bucketlist.query.filter_by(id=id).first()
+        if not bucketlist or (bucketlist.user_id != g.user.id):
+            abort(404, "bucketlist not found")
+        bucketlist.name = name
+        save(bucketlist)
+        msg = ("bucketlist ID: " + str(bucketlist.id) + " has been updated")
+        return {"message": msg}, 200
+
+    def delete(self, id=None):
+        if not id:
+            abort(400, "bad request")
+        bucketlist = Bucketlist.query.filter_by(id=id).first()
+        if not bucketlist or (bucketlist.user_id != g.user.id):
+            abort(404, "bucketlist not found")
+        delete(bucketlist)
+        msg = ("bucketlist : " + bucketlist.name + " has been deleted")
+        return {"message": msg}, 200
+
+
+class ItemAction(Resource):
+    decorators = [multi_auth.login_required]
```

Comment:
For purposes of convention, consider putting this decorator after the `__init__` function of your class.